### PR TITLE
Remove ConvertSegmentTask, HadoopConverterTask, and ConvertSegmentBackwardsCompatibleTask

### PR DESCRIPTION
**Pull request by jihoonson**
Fixes https://github.com/apache/incubator-druid/issues/6350.

### CODE REVIEW

**Comment by jihoonson** [comment]
Looks that the teamcity failure is unrelated.

File: docs/content/ingestion/misc-tasks.md [hunk 1]
```diff
@@ -4,68 +4,6 @@ layout: doc_page
 
 # Miscellaneous Tasks
 
-## Version Converter Task
```
Comment (jon-wei) [inline docs/content/ingestion/misc-tasks.md]: The "IndexSpec and Bitmap Types" section belong to "Indexing Service Convert Segment Task" and should also be removed

**Comment by nishantmonu51** [comment]
One use case i saw for ConvertSegemntTask was to convert bitmap indexes from consise to roaring by setting force flag and pointing the indexSpec to roaring, similar use case can be changing the compression strategies. 

Have we ruled out above cases ?

File: docs/content/ingestion/misc-tasks.md (hunk 1 above)
Comment (jihoonson) [reply docs/content/ingestion/misc-tasks.md]: Thanks. Removed.

**Comment by jihoonson** [comment]
@nishantmonu51 can we use reindexing (`IndexTask` + `IngestSegmentFirehose` or `CompactionTask`) for those use cases? I feel `ConvertSegmentTask` is for too specific use cases which are not very common.

**Comment by jihoonson** [comment]
I realized the teamcity failure legit. Fixed it.

**Comment by fjy** [comment]
@nishantmonu51 any more comments?

**Pull request merged**